File: applib/src/main/java/com/vic/applib/test/CompressTest.java
package com.vic.applib.test;

import com.vic.applib.utils.ConfigInfoKt;
import com.vic.lib.test.BaseTest;
import com.zxy.tiny.Tiny;
import com.zxy.tiny.callback.FileCallback;

/**
 * Created by Vic on 2017/7/17 0017.
 */

public class CompressTest extends BaseTest {
    @Override
    protected void doSubTest() {
//        final File file = new File(ConfigInfoKt.getAppDir() + "/4.png");
//        AsyncUtil.runThread(new RxRunnable<File>() {
//            @Override
//            public File run() {
//                return Compressor.getInstance().compressToFile(file,null);
//            }
//
//            @Override
//            public void onUI(File file) {
//                System.out.println("test compress");
//            }
//
//            @Override
//            public void onError(Throwable error) {
//
//            }
//        });


        Tiny.FileCompressOptions options = new Tiny.FileCompressOptions();
        options.outfile = ConfigInfoKt.getPicDir()+"/hh.png";
        Tiny.getInstance().source(ConfigInfoKt.getAppDir() + "/4.png").asFile().withOptions(options).compress(new FileCallback() {
            @Override
            public void callback(boolean isSuccess, String outfile) {
                //return the compressed file path
                System.out.println(outfile);
            }
        });

    }
}

File: applib/src/main/java/com/vic/applib/test/CompressTest.kt
package com.vic.applib.test

import android.graphics.Bitmap
import com.vic.applib.utils.LogUtil
import com.vic.applib.utils.appDir
import com.vic.applib.utils.compress.CompressorUtil
import com.vic.applib.utils.compress.OnCompressListener
import com.vic.lib.test.BaseTest
import java.io.File


/**
 * Created by Vic on 2017/7/3 0003.
 */
class CompressTest: BaseTest() {
    override fun doSubTest() {
        var file = File("$appDir/5.jpg")

        class List:OnCompressListener<File>{
            override fun onStart() {
                LogUtil.d("start compress 1")
            }

            override fun onSuccess(file: File?) {
                LogUtil.d("end compress 1")
            }

            override fun onError(e: Throwable?) {
            }

        }

        class ListenerBM:OnCompressListener<Bitmap>{
            override fun onStart() {
                LogUtil.d("start compress 1")
            }

            override fun onSuccess(t: Bitmap?) {
                if(t!=null) {
                    LogUtil.d("end compress 1")
                }else{
                    LogUtil.d("end compress 1 but bitmap is null")
                }

            }

            override fun onError(e: Throwable?) {
                LogUtil.d("end compress on error")
            }

        }
        CompressorUtil.CompressToBitmap(file,"test.jpg",ListenerBM())
    }

}